sdk/python/v1beta1/kubeflow/katib/api/katib_client.py

```diff
@@ -147,6 +147,7 @@ def tune(
         retain_trials: bool = False,
         packages_to_install: List[str] = None,
         pip_index_url: str = "https://pypi.org/simple",
+        resources: Optional[client.V1ResourceRequirements] = None,
```

Comment:
Should we try to be more Data Scientists friendly here ?

I believe, users might not be familiar how V1ResourceRequirements should be set and how [limits and requests work in Kubernetes](https://kubernetes.io/docs/concepts/configuration/manage-resources-containers/#requests-and-limits).

Maybe we could define this argument like `resources_per_trial`, and allow user to simply set num of CPU/GPU if they require it:
```py
resources_per_trial = {"cpu": "1", "gpu: "3", "memory": "1Gb"}
```
We can convert those resources to Kubernetes Pod containers `limits`.
Also, We can allow to define `resources_per_trial` as V1ResourceRequirements object for users who are familiar with `limit` and `requests` in Kubernetes.
WDYT @droctothorpe @johnugeorge @gaocegege @tenzen-y @anencore94 ?

Reply:
I wondered the same thing! 🙃

I went back and forth on it for a bit before concluding that wrapping / redefining the k8s interface would introduce operational overhead that we might prefer to avoid.

Simplifications comes at the cost of configurability. Users may want to configure both `requests` and `limits` discretely, for example.

The overwhelming majority of users won't need this, and the few who do are probably smart enough to instantiate a `V1ResourceRequirements` object.

Happy to refactor either way. Just sharing some thoughts on the topic.

Reply:
That make sense.
That is why I think that we can make argument `resources_per_trial: object` and accept it as dictionary: 
`{"gpu": "3"}` or as `V1ResourceRequirements`. Thanks to Python nature.

Reply:
Sounds good. Would just specify type as `resources_per_trial: Union[dict, V1ResourceRequirements, None] = None`.

Reply:
`Union` even better, Thanks! 
@gaocegege @johnugeorge @tenzen-y @a9p @anencore94 What are your thoughts on this ?

Reply:
SGTM on `resources_per_trial: Union[dict, V1ResourceRequirements, None] = None`.

Reply:
We made the requested changes @andreyvelich @tenzen-y. The one thing we weren't 100% sure about was what key to use for the gpu resource request. Internally, we rely on `nvidia.com/gpu`, and that's what we saw when going through the Python Kubernetes client issues. Just want to make sure that it works for your use case too.

Reply:
I think, we can keep `nvidia.com/gpu` as a default resource for GPU initially.
In the future, we can introduce `gpu_type` parameter to our KatibClient to users can modify this value (e.g. for AMD GPU `amd.com/gpu`).

Reply:
Sounds good. AMD users can pass `V1ResourceRequirements` anyway so it's not like they're blocked.